tests/js/client/shell/transaction/replication2_recovery/shell-transaction-replication2-recovery.js

```diff
@@ -0,0 +1,276 @@
+          replicatedStatePredicates.localKeyStatus(endpoint, dbn, shardId, "foo", true));
+      }
+
+      // Resume the dead server. Expect to read "foo" from it.
+      continueServerWait(leader);
+      syncShardsWithLogs(dbn);
+      replicatedLogsHelper.waitFor(
+        replicatedStatePredicates.localKeyStatus(
+          replicatedLogsHelper.getServerUrl(leader), dbn, shardId, "foo", true), 30);
+
+      // Try another transaction. This time expect it to work on all servers.
+      try {
+        trx = db._createTransaction({
+          collections: { write: c.name() }
+        });
+        tc = trx.collection(c.name());
+        tc.insert({ _key: "bar" });
+        trx.commit();
+      } catch (err) {
+        fail("Transaction failed with: " + JSON.stringify(err));
+      }
+
+      servers = Object.assign({}, ...participants.map(
+        (serverId) => ({ [serverId]: replicatedLogsHelper.getServerUrl(serverId) })));
+      for (const [_, endpoint] of Object.entries(servers)) {
+        replicatedLogsHelper.waitFor(
+          replicatedStatePredicates.localKeyStatus(endpoint, dbn, shardId, "bar", true));
+      }
+    },
+
+    /**
+     * This test is disabled because currently we can't properly stop the maintenance from deleting the shard
+     * on the newly added follower. When the core is constructed, the follower tries to create the new shard locally,
+     * but because the server is not listed as a participant for that shard in Plan/Collections, the maintenance figures
+     * the shard shouldn't exist locally.
+     */
+    DISABLED_testFollowerReplaceDuringTransaction: function () {
+      // Prepare the grounds for replacing a follower.
+      const logs = replicatedLogsHttpHelper.listLogs(coordinator, dbn).result;
+      const participants = logs[logId];
+      assertTrue(participants !== undefined);
+      const followers = participants.slice(1);
+      const oldParticipant = _.sample(followers);
+      const nonParticipants = _.without(replicatedLogsHelper.dbservers, ...participants);
+      const newParticipant = _.sample(nonParticipants);
+      const newParticipants = _.union(_.without(participants, oldParticipant), [newParticipant]).sort();
+
+      // Start a transaction.
+      let trx = db._createTransaction({
+        collections: {write: c.name()}
+      });
+      let tc = trx.collection(c.name());
+      tc.insert({_key: 'test1', value: 1});
+      tc.insert({_key: 'test2', value: 2});
+
+      // Replace the follower.
+      const result = replicatedStateHelper.replaceParticipant(dbn, logId, oldParticipant, newParticipant);
+      assertEqual({}, result);
+      replicatedLogsHelper.waitFor(() => {
+        const stateAgencyContent = replicatedStateHelper.readReplicatedStateAgency(dbn, logId);
+        return replicatedLogsHelper.sortedArrayEqualOrError(
+          newParticipants, Object.keys(stateAgencyContent.plan.participants).sort());
+      });
+
+      syncShardsWithLogs(dbn);
```

Comment:
I would like to add another "we cannot see test1 / test2 / test3 yet anywhere, before we trigger the commit.

Reply:
I added a TODO for that, this is a non-working test anyways, it's going to suffer modifications before we enable it.